WordPress/Classes/Utility/Media/MediaExternalExporter.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+
+/// A media asset protocol used to export media from external sources
+///
+protocol MediaExternalAsset {
+
+    /// The external URL
+    var URL: URL { get }
+    /// Asset name
+    var name: String { get }
+}
+
+/// Media export handling assets from external sources i.e.: Stock Photos
+///
+class MediaExternalExporter: MediaExporter {
+
+    public enum ExportError: MediaExportError {
+
+        case downloadError(NSError)
+
+        public var description: String {
+            switch self {
+            case .downloadError(let error):
+                return error.localizedDescription
+            }
+        }
+    }
+
+    var mediaDirectoryType: MediaDirectory = .uploads
+
+    let asset: MediaExternalAsset
+
+    init(externalAsset: MediaExternalAsset) {
+        asset = externalAsset
+    }
+
+    /// Downloads and export de external media asset
```

Comment:
This is almost irrelevant but still: _de_ should be _the_

Reply:
My Spanish unconscious mind 😱